src/rgw/rgw_coroutine.cc

```diff
@@ -1010,10 +1014,11 @@ bool RGWCoroutine::drain_children(int num_cr_left,
           int r = (*cb)(stack_id, ret);
           if (r < 0) {
             drain_status.ret = r;
+            drain_status.should_exit = true;
```

Comment:
@yehudasa this flag was added in https://github.com/ceph/ceph/pull/35511 but nothing set it to true. we need to set it here, so the `drain_with_cb()` macro can return the error from cb:
```c++
#define drain_with_cb(n, cb) \
  drain_status.init(); \
  yield_until_true(drain_children(n, cb)); \
  if (drain_status.should_exit) { \
    return set_cr_error(drain_status.ret); \                                   
  }
```